orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -1,581 +1,765 @@
+                        "Actual: nullptr.");
+      ORT_RETURN_IF_NOT(string_property->name(), "Checkpoint is invalid. Expected: Valid flatbuffer string property name. ",
+                        "Actual: nullptr.");
+      ORT_RETURN_IF_NOT(string_property->value(), "Checkpoint is invalid. Expected: Valid flatbuffer string property value. ",
+                        "Actual: nullptr.");
+      const std::string name = string_property->name()->str();
+      const std::string value = string_property->value()->str();
+      property_bag.AddProperty(name, value);
+    }
   }
 
   return Status::OK();
 }
 
-Status OrtLoadCustomPropertyInternal(const PathString& property_folder_path,
-                                     PropertyBag& property_bag) {
-  // Find custom property files.
-  std::vector<PathString> custom_property_filenames;
-  FilterFilesFromDirectory(
-      property_folder_path,
-      [&custom_property_filenames](const PathChar* filename) -> bool {
-        PathString filename_str = filename;
-        if (StringStartsWith(filename_str, k_property_root_prefix)) {
-          custom_property_filenames.push_back(filename_str);
-        }
-        return true;
-      });
-
-  if (custom_property_filenames.empty()) {
-    return Status::OK();
-  }
+#if !defined(ORT_MINIMAL_BUILD)
+/**
+ * @brief Load checkpoint from a checkpoint file to initializers in a model proto.
+ *
+ * @param checkpoint_path Path to the checkpoint file.
+ * @param model_proto Model proto to be populated.
+ * @return Status of the operation.
+ */
+Status ToModelProto(const PathString& checkpoint_path,
+                    ONNX_NAMESPACE::ModelProto& model_proto) {
+  InlinedVector<uint8_t> checkpoint_bytes;
+  ORT_RETURN_IF_ERROR(load::FromFile(checkpoint_path, checkpoint_bytes));
 
-  for (auto& property_file_path : custom_property_filenames) {
-    std::vector<ONNX_NAMESPACE::TensorProto> property_protos{};
-    auto property_file_full_path = ConcatPathComponent<PathChar>(property_folder_path, property_file_path);
-    LoadTensorProtoFromFile(property_file_full_path, property_protos, "[custom_property]");
+  const auto* fbs_checkpoint = fbs::GetCheckpoint(checkpoint_bytes.data());
+  ORT_RETURN_IF_NOT(fbs_checkpoint, "Checkpoint is invalid. Expected: Valid checkpoint flatbuffer. Actual: nullptr.");
 
-    for (auto& property_proto : property_protos) {
-      property_bag.AddProperty(property_proto);
-    }
+  const auto checkpoint_version = fbs_checkpoint->version();
+  ORT_RETURN_IF_NOT(IsCheckpointVersionSupported(checkpoint_version),
+                    "Checkpoint is invalid. Checkpoint version ", checkpoint_version, " is not supported.");
+
+  const auto* module_state = fbs_checkpoint->module_state();
+  if (nullptr == module_state) {
+    return Status::OK();
   }
 
-  return Status::OK();
-}
+  const auto* requires_grad_params = module_state->requires_grad();
```

Comment:
shall we rename `requires_grad_params` to  `requires_grad` in module_state?

Reply:
ok, sure. I can make this change.